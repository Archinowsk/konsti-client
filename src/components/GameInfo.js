/* @flow */
import React from 'react'
import { useTranslation } from 'react-i18next'
import timeFormatter from 'utils/timeFormatter'
import type { Game } from 'flow/game.flow'

export type Props = {
  game: Game,
}

const GameInfo = (props: Props) => {
  const { game } = props
  const { t } = useTranslation()

  if (!game) return <p />

  const getTags = () => {
    const tagsList = []

    if (game.noLanguage) {
      tagsList.push(<li key={'noLanguage'}>{t(`gameTags.noLanguage`)}</li>)
    }

    if (game.englishOk) {
      tagsList.push(<li key={'englishOk'}>{t(`gameTags.englishOk`)}</li>)
    }

    if (game.childrenFriendly) {
      tagsList.push(
        <li key={'childrenFriendly'}>{t(`gameTags.childrenFriendly`)}</li>
      )
    }

    if (game.ageRestricted) {
      tagsList.push(
        <li key={'ageRestricted'}>
          {t(`gameTags.ageRestricted`)} ({t(`gameTags.ageRestrictedLong`)})
        </li>
      )
    }

    if (game.beginnerFriendly) {
      tagsList.push(
        <li key={'beginnerFriendly'}>{t(`gameTags.beginnerFriendly`)}</li>
      )
    }

    if (game.intendedForExperiencedParticipants) {
      tagsList.push(
        <li key={'intendedForExperiencedParticipants'}>
          {t(`gameTags.intendedForExperiencedParticipants`)}
        </li>
      )
    }

    return tagsList
  }

  const getGenres = () => {
    let genresList = []
    if (game.genres) {
      genresList = game.genres.map(genre => (
        <li key={genre}>{t(`genre.${genre}`)}</li>
      ))
    }

    return genresList
  }

  const getStyles = () => {
    let stylesList = []
    if (game.styles) {
      stylesList = game.styles.map(style => (
        <li key={style}>{t(`gameStyle.${style}`)}</li>
      ))
    }

    return stylesList
  }

  const tagsList = getTags()
  const genresList = getGenres()
  const stylesList = getStyles()
  const formattedStartTime = timeFormatter.weekdayAndTime(game.startTime)
  const formattedEndTime = timeFormatter.timeOnly(game.endTime)

  return (
    <div>
      {game.title && (
        <div className='game-details-row'>
          <span className='game-details-title'>{t('gameInfo.title')}</span>
          {game.title}
        </div>
      )}
      {game.people && (
        <div className='game-details-row'>
          <span className='game-details-title'>{t('gameInfo.gamemaster')}</span>
          {game.people}
        </div>
      )}
      {genresList.length > 0 && (
        <div className='game-details-row'>
          <span className='game-details-title'>{t('gameInfo.genres')}</span>
          <ul>{genresList}</ul>
        </div>
      )}
      {tagsList.length > 0 && (
        <div className='game-details-row'>
          <span className='game-details-title'>{t('gameInfo.tags')}</span>
          <ul>{tagsList}</ul>
        </div>
      )}
      {game.mins && (
        <div className='game-details-row'>
          <span className='game-details-title'>{t('gameInfo.runTime')}</span>
          {formattedStartTime} - {formattedEndTime} ({game.mins / 60}{' '}
          {t('hours')})
        </div>
      )}
      {game.description && (
        <div className='game-details-row'>
          <span className='game-details-title'>
            {t('gameInfo.description')}
          </span>
          {game.description}
        </div>
      )}
      {game.gameSystem && (
        <div className='game-details-row'>
          <span className='game-details-title'>{t('gameInfo.gamesystem')}</span>
          {game.gameSystem}
        </div>
      )}
      {stylesList.length > 0 && (
        <div className='game-details-row'>
          <span className='game-details-title'>{t('gameInfo.gameStyle')}</span>
          <ul>{stylesList}</ul>
        </div>
      )}
      {game.location && (
        <div className='game-details-row'>
          <span className='game-details-title'>{t('gameInfo.location')}</span>
          {game.location}
        </div>
      )}
      {game.minAttendance && game.maxAttendance && (
        <div className='game-details-row'>
          <span className='game-details-title'>
            {t('gameInfo.numberOfPlayers')}
          </span>
          {game.minAttendance} - {game.maxAttendance}
        </div>
      )}
    </div>
  )
}

export default GameInfo
