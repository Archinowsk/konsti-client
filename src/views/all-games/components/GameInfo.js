/* @flow */
import React, { Fragment } from 'react'
import { useTranslation } from 'react-i18next'
import { timeFormatter } from 'utils/timeFormatter'
import type { Game } from 'flow/game.flow'
import type { StatelessFunctionalComponent, Element } from 'react'

export type Props = {|
  game: Game,
|}

export const GameInfo: StatelessFunctionalComponent<Props> = (
  props: Props
): Element<'div'> => {
  const { game } = props
  const { t } = useTranslation()

  if (!game) return <div className='game-details' />

  const getTags = () => {
    const tagsList = []

    if (game.noLanguage) {
      tagsList.push(`noLanguage`)
    }

    if (game.englishOk || game.tags.includes('in-english')) {
      tagsList.push(`englishOk`)
    }

    if (game.childrenFriendly) {
      tagsList.push(`childrenFriendly`)
    }

    if (game.ageRestricted) {
      tagsList.push(`ageRestricted`)
    }

    if (game.beginnerFriendly) {
      tagsList.push(`beginnerFriendly`)
    }

    if (game.intendedForExperiencedParticipants) {
      tagsList.push(`intendedForExperiencedParticipants`)
    }

    return tagsList.map(tag => t(`gameTags.${tag}`)).join(', ')
  }

  const getGenres = (genresList: $ReadOnlyArray<string>) => {
    return genresList.map(genre => t(`genre.${genre}`)).join(', ')
  }

  const getStyles = (styles: $ReadOnlyArray<string>) => {
    return styles.map(style => t(`gameStyle.${style}`)).join(', ')
  }

  const tagsList = getTags()
  const formattedStartTime = timeFormatter.weekdayAndTime(game.startTime)
  const formattedEndTime = timeFormatter.time(game.endTime)

  return (
    <div className='game-details'>
      {game.title && (
        <div className='game-details-row game-details-row-with-subtext'>
          <h3 className='game-details-title'>{game.title}</h3>
        </div>
      )}

      {game.shortDescription && (
        <div className='game-details-row game-details-subtext'>
          <span className='game-details-text-indent italic'>
            {game.shortDescription}
          </span>
        </div>
      )}

      {game.people && (
        <div className='game-details-row game-details-row-with-gap'>
          <span className='game-details-text-indent'>{game.people}</span>
        </div>
      )}

      {game.revolvingDoor && (
        <Fragment>
          <div className='game-details-row game-details-row-with-subtext'>
            <span className='game-details-title'>
              {t('gameInfo.revolvingDoor')}
            </span>
          </div>

          <div className='game-details-row game-details-subtext game-details-row-with-gap'>
            <span className='game-details-text-indent'>
              {t('gameInfo.revolvingDoorDescription')}
            </span>
          </div>
        </Fragment>
      )}

      {game.ageRestricted && (
        <Fragment>
          <div className='game-details-row game-details-row-with-subtext'>
            <span className='game-details-title'>
              {t('gameTags.ageRestrictedTitle')}
            </span>
          </div>

          <div className='game-details-row game-details-subtext game-details-row-with-gap'>
            <span className='game-details-text-indent'>
              {t('gameTags.ageRestrictedLong')}
            </span>
          </div>
        </Fragment>
      )}

      {!!game.mins && (
        <div className='game-details-row'>
          <span className='game-details-title game-details-two-columns'>
            {t('gameInfo.runTime')}
          </span>
          <span className='game-details-value'>
            {formattedStartTime} - {formattedEndTime} ({game.mins / 60}{' '}
            {t('hours')})
          </span>
        </div>
      )}

      {game.genres.length > 0 && (
        <div className='game-details-row'>
          <span className='game-details-title game-details-two-columns'>
            {t('gameInfo.genres')}
          </span>
          <span className='game-details-value'>{getGenres(game.genres)}</span>
        </div>
      )}

      {tagsList.length > 0 && (
        <div className='game-details-row'>
          <span className='game-details-title game-details-two-columns'>
            {t('gameInfo.tags')}
          </span>
          <span className='game-details-value'>{t(tagsList)}</span>
        </div>
      )}

      {game.description && (
        <Fragment>
          <div className='game-details-row'>
            <span className='game-details-title'>
              {t('gameInfo.description')}
            </span>
          </div>
          <div className='game-details-row game-details-row-with-gap'>
            <span>{game.description}</span>
          </div>
        </Fragment>
      )}

      {game.gameSystem && (
        <div className='game-details-row'>
          <span className='game-details-title game-details-two-columns'>
            {t('gameInfo.gamesystem')}
          </span>
          <span className='game-details-value'>{game.gameSystem}</span>
        </div>
      )}

      {game.styles.length > 0 && (
        <div className='game-details-row'>
          <span className='game-details-title game-details-two-columns'>
            {t('gameInfo.gameStyle')}
          </span>
          <span className='game-details-value'>{getStyles(game.styles)}</span>
        </div>
      )}

      {game.location && (
        <div className='game-details-row'>
          <span className='game-details-title game-details-two-columns'>
            {t('gameInfo.location')}
          </span>
          <span className='game-details-value'>{game.location}</span>
        </div>
      )}

      {!!game.minAttendance && !!game.maxAttendance && (
        <div className='game-details-row'>
          <span className='game-details-title game-details-two-columns'>
            {t('gameInfo.numberOfPlayers')}
          </span>
          <span className='game-details-value'>
            {game.minAttendance} - {game.maxAttendance}
          </span>
        </div>
      )}
    </div>
  )
}
