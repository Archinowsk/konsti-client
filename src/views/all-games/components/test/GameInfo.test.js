// @flow
import React from 'react'
import { shallow } from 'enzyme'
import { testGame } from 'test/test-data/testGame'
import { GameInfo } from '../GameInfo'
import type { Props } from '../GameInfo'

describe('GameInfo', () => {
  it('should render correctly', () => {
    const props: Props = { game: testGame }
    const component = shallow(<GameInfo {...props} />)
    expect(component).toMatchSnapshot()
  })
})
