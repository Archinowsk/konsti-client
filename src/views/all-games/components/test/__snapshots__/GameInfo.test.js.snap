// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameInfo should render correctly 1`] = `
<div
  className="game-details"
>
  <div
    className="game-details-row game-details-row-with-subtext"
  >
    <h3
      className="game-details-title"
    >
      Test game
    </h3>
  </div>
  <div
    className="game-details-row game-details-subtext"
  >
    <span
      className="game-details-text-indent italic"
    >
      Short description
    </span>
  </div>
  <div
    className="game-details-row game-details-row-with-gap"
  >
    <span
      className="game-details-text-indent"
    >
      Test GM
    </span>
  </div>
  <div
    className="game-details-row game-details-row-with-subtext"
  >
    <span
      className="game-details-title"
    >
      gameInfo.revolvingDoor
    </span>
  </div>
  <div
    className="game-details-row game-details-subtext game-details-row-with-gap"
  >
    <span
      className="game-details-text-indent"
    >
      gameInfo.revolvingDoorDescription
    </span>
  </div>
  <div
    className="game-details-row"
  >
    <span
      className="game-details-title game-details-two-columns"
    >
      gameInfo.runTime
    </span>
    <span
      className="game-details-value"
    >
      <span>
        Saturday 19:00
         - 
        23:00
         
      </span>
      <span
        className="no-wrap"
      >
        (
        4 hours
        )
      </span>
    </span>
  </div>
  <div
    className="game-details-row"
  >
    <span
      className="game-details-title game-details-two-columns"
    >
      gameInfo.genres
    </span>
    <span
      className="game-details-value"
    >
      <span
        key="fantasy"
      >
        <span
          className="no-wrap"
        >
          genre.fantasy
        </span>
        <span>
          , 
        </span>
      </span>
      <span
        key="war"
      >
        <span
          className="no-wrap"
        >
          genre.war
        </span>
        <span>
          , 
        </span>
      </span>
      <span
        key="exploration"
      >
        <span
          className="no-wrap"
        >
          genre.exploration
        </span>
        <span>
          , 
        </span>
      </span>
      <span
        key="mystery"
      >
        <span
          className="no-wrap"
        >
          genre.mystery
        </span>
        <span>
          , 
        </span>
      </span>
      <span
        key="drama"
      >
        <span
          className="no-wrap"
        >
          genre.drama
        </span>
        <span />
      </span>
    </span>
  </div>
  <div
    className="game-details-row"
  >
    <span
      className="game-details-title"
    >
      gameInfo.description
    </span>
  </div>
  <div
    className="game-details-row game-details-row-with-gap"
  >
    <span>
      Test game description
    </span>
  </div>
  <div
    className="game-details-row"
  >
    <span
      className="game-details-title game-details-two-columns"
    >
      gameInfo.gamesystem
    </span>
    <span
      className="game-details-value"
    >
      Test gamesystem
    </span>
  </div>
  <div
    className="game-details-row"
  >
    <span
      className="game-details-title game-details-two-columns"
    >
      gameInfo.gameStyle
    </span>
    <span
      className="game-details-value"
    >
      <span
        key="serious"
      >
        <span
          className="no-wrap"
        >
          gameStyle.serious
        </span>
        <span>
          , 
        </span>
      </span>
      <span
        key="story_driven"
      >
        <span
          className="no-wrap"
        >
          gameStyle.story_driven
        </span>
        <span>
          , 
        </span>
      </span>
      <span
        key="character_driven"
      >
        <span
          className="no-wrap"
        >
          gameStyle.character_driven
        </span>
        <span />
      </span>
    </span>
  </div>
  <div
    className="game-details-row"
  >
    <span
      className="game-details-title game-details-two-columns"
    >
      gameInfo.location
    </span>
    <span
      className="game-details-value"
    >
      Test location
    </span>
  </div>
  <div
    className="game-details-row"
  >
    <span
      className="game-details-title game-details-two-columns"
    >
      gameInfo.numberOfPlayers
    </span>
    <span
      className="game-details-value"
    >
      2
       - 
      4
    </span>
  </div>
</div>
`;
